### PR TITLE
feat: hybridse-sdk SqlEngine support multiple databases

### LINKED ISSUE #472: Support passing multiple database for SqlEngine in hybridse-sdk
Now we can only pass one database to SqlEngine but we may should support passing multiple databases in catalog for core SQL engine.

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Resolve https://github.com/4paradigm/OpenMLDB/issues/472 .

### CODE REVIEW

File: hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java [hunk 1]
```diff
@@ -83,20 +107,22 @@ public static EngineOptions createDefaultEngineOptions() {
      * Initialize engine with given sql, database and specified engine options.
      *
      * @param sql query sql string
-     * @param database query on the database
+     * @param databases query on the databases
      * @param engineOptions query engine options
      * @throws UnsupportedHybridSeException throw when query unsupported or has syntax error
      */
-    public void initilize(String sql, TypeOuterClass.Database database, EngineOptions engineOptions)
+    public void initilize(String sql, List<TypeOuterClass.Database> databases, EngineOptions engineOptions)
```
Comment (jingchen2222) [inline hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java]: Option 1: add @param default_database_name to **explicitly** specific default database  for query like `SELECT col1 from t1;`
Option 2: use 1st database of databases as the default database. But I think users will be confused about the rules even we clarify the rules in user manual docs.  Besides, the query statement might not use the DEFAULT database.
Prefer using option 1.

File: hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java [hunk 2]
```diff
@@ -83,20 +107,22 @@ public static EngineOptions createDefaultEngineOptions() {
...
         session = new BatchRunSession();
-        catalog.AddDatabase(database);
+        for (TypeOuterClass.Database database: databases) {
+            catalog.AddDatabase(database);
+        }
         engine = new Engine(catalog, options);
 
         BaseStatus status = new BaseStatus();
-        boolean ok = engine.Get(sql, database.getName(), session, status);
+        boolean ok = engine.Get(sql, databases.get(0).getName(), session, status);
```
Comment (tobegit3hub) [inline hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java]: Do not use default database name from the first database

File: hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java (hunk 1 above)
Comment (tobegit3hub) [inline hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java]: add the parameter of database name(Type String)

**Review by zekai427 — APPROVED** [review]

**Review by jingchen2222 — APPROVED** [review]
LGTM

File: hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java (hunk 1 above)
Comment (tobegit3hub) [reply hybridse/java/hybridse-sdk/src/main/java/com/_4paradigm/hybridse/sdk/SqlEngine.java]: The default database name is required for multiple databases.

**Pull request merged**